### ISSUE TITLE
RuleRequiredForAtLeast1Property Error in published Blazor.Server App

### ISSUE THREAD

**Issue by huncu16** [eXpandFramework/eXpand#1056]
Hello eXpand Team,

I converted our XAF app from .Net to .Net.Core Framework and create its blazor server app all works fine while starting it from Visual Studio Debug or Relase mode. But if I publish it and start server in IIS or running MyAppName.Blazor.Server.exe i get the following error before login screen display 

Error: System.InvalidOperationException: Unable to create a model node for the 'RuleRequiredForAtLeast1Property' rule. The rule type is unregistered. Use the ValidationRulesRegistrator class to register the rule on module setup.

after i get this error
I've be sure about add Xpand.Persistent.Base refence is ok then i've registered it at MyApp.Module.Module.cs in Application_SetupComplate method before initialize it wia code below

ValidationRulesRegistrator.RegisterRule(module.ModuleManager, typeof(RuleRequiredForAtLeast1Property), typeof(RuleBasePropertiesLogic));

but nothing changed in published app. It still works fine in debug or rease started with VS.

I put the screen shots below for callstack

What am i missing?

![Clipboard-File-1](https://github.com/eXpandFramework/eXpand/assets/39297744/8d73f162-89b8-48bb-acce-4d74444842b1)
![Clipboard-File-2](https://github.com/eXpandFramework/eXpand/assets/39297744/adb6ec7e-1b5a-4fff-be5a-fcd0d74a5563)

Thanx for answers.

**Comment by apobekiaris** [comment]
hard to tell without experementing the most obvious is to blame your publication, so my guess is that RuleRequiredForAtLeast1Property is actually in a different versioned assemby

**Comment by huncu16** [comment]
Yes it cannot be publication while relase.exe giving same error and ican build and run in VS. I've checked all versions are same 22.1.302.2

**Comment by huncu16** [comment]
Tolis, Hello again, 
After been sure all versions are same then I updated my solution from 22.1 to 23.2 (I've planned it after successfull publish in 22.1 but set its plan to early because of this problem) i still get the same error. But there is one diffenrence is now i cannot run my project vi Visual Studio too.

**Comment by apobekiaris** [comment]
I am sorry without repro solution cannot help much

**Comment by expand** [comment]
Closing issue for age. Feel free to reopen it at any time.

.Thank you for your contribution.
